kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -230,6 +280,7 @@ class LineSearchStrategy
      * @param pNewBuilderAndSolver The builder and solver employed
      * @param ThisParameters Settings used in the strategy
      */
+    KRATOS_DEPRECATED_MESSAGE("Constructor deprecated, please use the constructor without linear solver")
```

Comment:
the CTors with Parameters are quite new, I think we can directly take the lin solver argument out without backwards compatibility